infra/gcp/istio-prow-build/cluster-prow-arm-private.tf

```diff
@@ -0,0 +1,250 @@
+# Prow private cluster is the private mirror of the "prow-arm" cluster.
+# Access to this is restricted to only private infrastructure.
+resource "google_container_cluster" "prow_arm_private" {
+  addons_config {
+    dns_cache_config {
+      enabled = false
+    }
+
+    gce_persistent_disk_csi_driver_config {
+      enabled = true
+    }
+
+    horizontal_pod_autoscaling {
+      disabled = false
+    }
+
+    http_load_balancing {
+      disabled = false
+    }
+
+    network_policy_config {
+      disabled = true
+    }
+  }
+
+  binary_authorization {
+    evaluation_mode = "DISABLED"
+  }
+
+  cluster_autoscaling {
+    enabled = false
+  }
+
+  database_encryption {
+    state = "DECRYPTED"
+  }
+
+  datapath_provider         = "LEGACY_DATAPATH"
+  default_max_pods_per_node = 110
+
+  default_snat_status {
+    disabled = false
+  }
+
+  enable_shielded_nodes = true
+
+  ip_allocation_policy {
+    cluster_ipv4_cidr_block  = "10.52.0.0/14"
+    services_ipv4_cidr_block = "10.56.0.0/20"
+  }
+
+  location = "us-central1-f"
+
+  logging_config {
+    enable_components = ["SYSTEM_COMPONENTS", "WORKLOADS"]
+  }
+
+  master_auth {
+    client_certificate_config {
+      issue_client_certificate = false
+    }
+  }
+
+  monitoring_config {
+    enable_components = ["SYSTEM_COMPONENTS"]
+
+    managed_prometheus {
+      enabled = false
+    }
+  }
+
+  name    = "prow-arm-private"
+  network = "projects/istio-prow-build/global/networks/default"
+
+  network_policy {
+    enabled  = false
+    provider = "PROVIDER_UNSPECIFIED"
+  }
+
+  networking_mode = "VPC_NATIVE"
+
+  node_version = "1.25.8-gke.500"
```

Comment:
is there a reason arm is different than private? 
node_version = "1.24.12-gke.500"

Reply:
Was created later. For now this is matching the existing state, once we merge this it will be easier to safely make changes like making them the same version

Reply:
Actually ARM is only 1.25+ I think